### PR TITLE
Make it build with ghc 9.10

### PR DESCRIPTION
**Pull request by erikd**
~This is a WIP mainly intended to make the Plutus maintainers aware of the fact that I am working on this.~

~This cannot be merged until the (currently ~17~ 16) SRPs are fixed upstream or fully documented.~

~**The fact that the Nix evals are failing can be ignored for now.~
~The Nix failures are due to missing SHA hashes for the SRPs which need to be removed before this PR can be merged and are likely to be changing frequenty.**~

Ready for review. No Changelog Required.

### CODE REVIEW

File: plutus-core/plutus-core.cabal [hunk 1]
```diff
@@ -286,7 +286,7 @@ library
     , bytestring
     , bytestring-strict-builder
     , cardano-crypto
-    , cardano-crypto-class        ^>=2.1.2
+    , cardano-crypto-class        ^>= 2.2
```
Comment (erikd) [inline plutus-core/plutus-core.cabal:289]: Any reason not to do this? Plutus seems to compile with this change.

Comment (erikd) [reply plutus-core/plutus-core.cabal]: Reason not to do this is because `cardano-ledger` does not compile with the new version of `cardano-crypto-class`.

**Comment by erikd** [comment]
I have removed all the SRPs, but can not longer push to this branch.

**Comment by erikd** [comment]
Please check the last two commits on this PR. One test was disabled and 3 golden output changes were accepted.

File: plutus-core/cost-model/budgeting-bench/Generators.hs [hunk 2]
```diff
@@ -10,7 +11,9 @@ import Control.Monad
 import Data.Bits
 import Data.ByteString (ByteString)
 import Data.Int (Int64)
+#if !MIN_VERSION_base(4,20,0)
 import Data.List (foldl')
+#endif
```
Comment (effectfully) [inline plutus-core/cost-model/budgeting-bench/Generators.hs:16]: Any way we could do this without the CPP?

File: plutus-core/plutus-core/src/PlutusCore/Name/UniqueMap.hs [hunk 3]
```diff
@@ -23,7 +24,9 @@ module PlutusCore.Name.UniqueMap (
 import Control.Lens (view)
 import Control.Lens.Getter ((^.))
 import Data.Coerce (Coercible, coerce)
-import Data.Foldable (foldl')
+#if ! MIN_VERSION_base(4,20,0)
+import Data.List (foldl')
+#endif
```
Comment (effectfully) [inline plutus-core/plutus-core/src/PlutusCore/Name/UniqueMap.hs:29]: Same question.

File: plutus-tx/src/PlutusTx/Builtins/HasOpaque.hs [hunk 4]
```diff
@@ -22,6 +23,10 @@ import Data.String (IsString (..))
 import Data.Text qualified as Text
 import GHC.Magic qualified as Magic
 import Prelude qualified as Haskell (String)
+#if MIN_VERSION_base(4,20,0)
+import Prelude (type (~))
+#endif
```
Comment (effectfully) [inline plutus-tx/src/PlutusTx/Builtins/HasOpaque.hs:28]: God, why do they make us go through this.

**Review by effectfully — COMMENTED** [review]
All the new `foldl'`-related CPP is pretty annoying, I'll take a look to see if we can avoid it. Maybe by pulling it into `PlutusPrelude`? I'll check it.

File: plutus-core/cost-model/budgeting-bench/Generators.hs (hunk 2 above)
Comment (erikd) [reply plutus-core/cost-model/budgeting-bench/Generators.hs:16]: I removed it somewhere else so will do the same here and the other places.

File: plutus-tx/src/PlutusTx/Builtins/HasOpaque.hs (hunk 4 above)
Comment (erikd) [reply plutus-tx/src/PlutusTx/Builtins/HasOpaque.hs:28]: I have removed all `CPP` that I added other than that one.

File: plutus-core/cost-model/budgeting-bench/Generators.hs (hunk 2 above)
Comment (erikd) [reply plutus-core/cost-model/budgeting-bench/Generators.hs]: Fixed.

**Review by effectfully — APPROVED** [review]
LGTM

**Pull request merged**